riptide/ollama_heal.py

```
@@ -0,0 +1,148 @@
+def is_systemd_available() -> bool:
+    """Check if systemd user session is functional."""
+    return os.environ.get("DBUS_SESSION_BUS_ADDRESS") is not None and os.path.exists(
+        f"/run/user/{os.getuid()}"
+    )
+
+
+def is_systemd_service_loaded() -> bool:
+    """Check if the systemd user service exists and is loaded."""
+    if not is_systemd_available():
+        return False
+    result = subprocess.run(
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)